### PR TITLE
Add Helm Chart for Keycloak

### PR DESCRIPTION
**Pull request by Feleuxens**

### CODE REVIEW

File: templates/statefulset.yaml [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+      labels:
+        {{- range $key, $val := .Values.podLabels }}
+        {{- printf "%s: %s" $key ($val | quote) | nindent 8 }}
+        {{- end }}
+        {{- include "keycloak.selectorLabels" . | nindent 8 }}
+    spec:
+      containers:
+      - name: keycloak
+        {{- if .Values.image.digest}}
+        image: "{{ .Values.image.repository }}@{{ .Values.image.digest }}"
```
Comment (ingokuba) [inline templates/statefulset.yaml:41]: isn't the tag missing here if you want to pull e.g. `quay.io/keycloak/keycloak:22.0.1@sha256:9870...`

see https://stackoverflow.com/a/63204994/9710727

**Review by ingokuba — CHANGES_REQUESTED** [review]

File: README.md [hunk 2]
```diff
@@ -1,2 +1,119 @@
-# keycloak-helm-chart
+# Keycloak Helm Chart
+
 Helm Chart for Keycloak by intension
+
+## Post install jobs
+
+This chart offers the possibility to define jobs that run after the installation. To enforce that it runs after the pod is ready, either use the `--wait` option from `helm install` or define a init container that waits for pod to be ready.
```
[the remark this answers is not in the document]
Comment (Feleuxens) [reply README.md:7]: Done https://github.com/intensiongmbh/keycloak-helm-chart/commit/3ef46e73155274a5e11d91aa4afccea67ceaf5a4

File: README.md [hunk 3]
```diff
@@ -1,2 +1,119 @@
-# keycloak-helm-chart
+# Keycloak Helm Chart
+
 Helm Chart for Keycloak by intension
+
```
[the remark this answers is not in the document]
Comment (Feleuxens) [reply README.md:4]: Done https://github.com/intensiongmbh/keycloak-helm-chart/commit/3ef46e73155274a5e11d91aa4afccea67ceaf5a4

File: templates/statefulset.yaml (hunk 1 above)
Comment (Feleuxens) [reply templates/statefulset.yaml:41]: Apparently, if a digest is specified, the tag is ignored. `docker image pull quay.io/keycloak/keycloak:notexistingtag@sha256:5b872e8...` works regardless of which tag is used.

File: README.md [hunk 4]
```diff
@@ -1,2 +1,119 @@
...
+| `autoscaling.maxReplicas`                       | Maximum number of replicas to scale up                                                                                            | `100`           |
+| `autoscaling.targetCPUUtilizationPercentage`    | Target CPU utilization percentage                                                                                                 | `80`            |
+| `autoscaling.targetMemoryUtilizationPercentage` | Target Memory utilization percentage                                                                                              | `nil`           |
+| `autoscaling.labels`                            | Labels for the HPA                                                                                                                | `{}`            |
+| `postinstalljobs`                               | Define a list of jobs to be executed after the installation                                                                       | `[]`            |
```
[the remark this answers is not in the document]
Comment (Feleuxens) [reply README.md]: https://github.com/intensiongmbh/keycloak-helm-chart/pull/1/commits/38e03e29da5590fc86a9b50483f6b51921b7b370
https://github.com/intensiongmbh/keycloak-helm-chart/pull/1/commits/c858878a9eff66e23eb8a8ba2632f05a5f832537

File: README.md [hunk 5]
```diff
@@ -1,2 +1,119 @@
...
+
+### Database parameters
+
+| Name                         | Description                                                                        | Value |
+| ---------------------------- | ---------------------------------------------------------------------------------- | ----- |
+| `database.existingSecret`    | Use an existing secret for database password. This takes precedence over password. | `""`  |
+| `database.existingSecretKey` |                                                                                    | `""`  |
```
[the remark this answers is not in the document]
Comment (Feleuxens) [reply README.md]: https://github.com/intensiongmbh/keycloak-helm-chart/pull/1/commits/9eee5fd8074ca7d68ac8f28f988e3d771937514d

File: README.md [hunk 6]
```diff
@@ -1,2 +1,119 @@
...
+### Keycloak parameters
+
+| Name                | Description                                                           | Value                       |
+| ------------------- | --------------------------------------------------------------------- | --------------------------- |
+| `image.repository`  | The image repository for the Keycloak container                       | `quay.io/keycloak/keycloak` |
+| `image.pullPolicy`  | Pull policy of the image used for Keycloak                            | `IfNotPresent`              |
+| `image.tag`         | Overrides the image tag whose default is the chart appVersion.        | `22.0.1`                    |
+| `image.digest`      | Overrides the image tag with a specific digest                        | `""`                        |
+| `hostAliases`       | Mapped IPs and hostnames that will be injected into Pod's hosts files | `[]`                        |
+| `http.relativePath` | For compatibility reason this could be set to "/auth"                 | `""`                        |
```
[the remark this answers is not in the document]
Comment (Feleuxens) [reply README.md:32]: Not necessarily, there are options like proxy, metrics and health that set environment variables for Keycloak and the other options are setting for the StatefulSet.

File: Chart.yaml [hunk 7]
```diff
@@ -1,6 +1,6 @@
 apiVersion: v2
 name: keycloak
-description: A Helm chart for Kubernetes
+description: Keycloak - SSO now within Kubernetes
```
Comment (ingokuba) [inline Chart.yaml:3]: Weiß ich ja nicht... aber mir fällt nix besseres ein

File: README.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (ingokuba) [reply README.md:33]: ah okey, can you write that above the table please?

[the remark this answers is not in the document]
Comment (Feleuxens) [reply README.md]: https://github.com/intensiongmbh/keycloak-helm-chart/pull/1/commits/ef2719bb36323cbdc5bb59b27adda32ddb975241

**Pull request merged**